src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,112 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private int currentCapacity;
+    private int userCapacity;
+    private Node<K, V>[] arrayOfElement;
+
+    public MyHashMap() {
+        size = 0;
+        currentCapacity = DEFAULT_CAPACITY;
+        arrayOfElement = new Node[DEFAULT_CAPACITY];
+    }
+
+    public MyHashMap(int userCapacity) {
+        if (userCapacity <= 0) {
+            throw new RuntimeException("Array size is incorrect");
+        }
+        arrayOfElement = new Node[userCapacity];
+        currentCapacity = userCapacity;
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        checkSize();
+        putIntoArray(key, value);
+    }
 
+    private void putIntoArray(K key, V value) {
+        if (arrayOfElement[getIndexElementArray(key)] == null) {
+            arrayOfElement[getIndexElementArray(key)] = new Node<>(key, value, null);
+            size++;
+            return;
+        } else {
+            Node<K, V> existNode = arrayOfElement[getIndexElementArray(key)];
+            for (Node<K, V> i = existNode; i != null; i = i.next) {
+                if (Objects.equals(i.key, key)) {
+                    i.value = value;
+                    break;
+                } else if (i.next == null) {
+                    Node<K, V> newNode = new Node<>(key, value, null);
+                    i.next = newNode;
+                    size++;
+                    break;
+                }
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
```

Comment:
this check is redundant, cause you won't enter you for loop, if it's empty, you can just remove it